native/src/main/java/io/ballerina/stdlib/jsonpath/BJsonProvider.java

```diff
@@ -0,0 +1,271 @@
+             throw new UnsupportedOperationException();
+         }
+         BArray arrayNode = toJsonArray(array);
+         if (index == arrayNode.size()) {
+             arrayNode.append(createJsonElement(newValue));
+         } else {
+             arrayNode.add(index, createJsonElement(newValue));
+         }
+     }
+ 
+     public Object getMapValue(Object obj, String key) {
+         if (!isMap(obj)) {
+             throw new UnsupportedOperationException();
+         }
+         BMap jsonObject = toJsonObject(obj);
+         if (!jsonObject.containsKey(StringUtils.fromString(key))) {
+             return UNDEFINED;
+         } else {
+             return unwrap(jsonObject.get(StringUtils.fromString(key)));
+         }
+     }
+ 
+     public void setProperty(Object obj, Object key, Object value) {
+         if (isMap(obj)) {
+             toJsonObject(obj).put(key.toString(), createJsonElement(value));
+         } else {
+             BArray array = toJsonArray(obj);
+             long index;
+             if (key != null) {
+                 index = key instanceof Integer ? (Integer) key : Integer.parseInt(key.toString());
+             } else {
+                 index = array.size();
+             }
+ 
+             if (index == array.size()) {
+                 array.append(createJsonElement(value));
+             } else {
+                 array.add(index, createJsonElement(value));
+             }
+         }
+     }
+ 
+     public void removeProperty(Object obj, Object key) {
+         if (isMap(obj)) {
+             toJsonObject(obj).remove(key.toString());
+         } else {
+             BArray array = toJsonArray(obj);
+             int index = key instanceof Integer ? (Integer) key : Integer.parseInt(key.toString());
+             array.shift(index);
+         }
+     }
+ 
+     public boolean isMap(Object obj) {
+         return (obj instanceof BMap);
+     }
+ 
+     public Collection<String> getPropertyKeys(Object obj) {
+         List<String> keys = new ArrayList<>();
+         for (Object entry : toJsonObject(obj).getKeys()) {
+             keys.add(entry.toString());
+         }
+         return keys;
+     }
+ 
+     public int length(Object obj) {
+         if (isArray(obj)) {
+             return toJsonArray(obj).size();
+         } else if (isMap(obj)) {
+             return toJsonObject(obj).entrySet().size();
+         } else {
+             if (isJsonPrimitive(obj)) {
+                 return obj.toString().length();
+             }
+         }
+         throw new JsonPathException("length operation can not applied to " + (obj != null ? obj.getClass().getName()
+                 : "null"));
+     }
+ 
+     public Iterable<?> toIterable(Object obj) {
+         BArray arr = toJsonArray(obj);
+         List<Object> values = new ArrayList<>(arr.size());
+         for (int i = 0; i < arr.size(); i++) {
+             values.add(unwrap(arr.get(i)));
+         }
+         return values;
```

Comment:
Can't we make this also compact. WDYT?
```suggestion
         return Arrays.stream(toJsonArray(obj).getValues())
                .map(this::unwrap)
                .collect(Collectors.toList());
```

Reply:
When we using this, there is a index out of range issue in several tests.
So i think stick to the original implementation will be better, since it create a array list with exact length